### PR TITLE
Trim unneeded CUDA arch feedstocks (add 2 needed ones)

### PR DESCRIPTION
**Pull request by jakirkham**
Trims unneeded CUDA feedstocks from the arch migration (see xref'd PR for context). Adds 2 needed feedstocks that were missed previously

xref: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2699

cc @isuruf

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/migrations/arch_rebuild.txt [hunk 1]
```diff
@@ -509,35 +509,12 @@ rmm
...
 libcugraph
-mandrake
-meld
-nnpops
 nvidia-apex
 omniscidb
-openmm-plumed
-openmm-torch
-prismatic_split
+openmm
```
Comment (jakirkham) [inline recipe/migrations/arch_rebuild.txt:518]: `openmm` would like to build on other architectures ( https://github.com/conda-forge/openmm-feedstock/issues/8 ) and have discussed this with them before

File: recipe/migrations/arch_rebuild.txt [hunk 2]
```diff
@@ -509,35 +509,12 @@ rmm
...
-pytorch_cluster
-pytorch_sparse
-pytorch_spline_conv
-seekr2_openmm_plugin
-timemory
-torch-scatter
-torchsparse
-tsnecuda
-tvm-py
+xgboost
```
Comment (jakirkham) [inline recipe/migrations/arch_rebuild.txt:520]: We also work on this package and have needs for ARM

**Comment by jakirkham** [comment]
Commented on the 2 added ones above for context. The rest that were kept were already needed before. Dropped everything that wasn't needed.

**Review by isuruf — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Isuru! 😄

**Pull request merged**